src/storage/src/source/postgres/replication.rs

```diff
@@ -466,18 +488,36 @@ async fn raw_stream<'a>(
         Err(err) => return Err(err.into()),
     };
 
+    // According to the documentation [1] we must check that the slot LSN matches our
+    // expectations otherwise we risk getting silently fast-forwarded to a future LSN. In order
+    // to avoid a TOCTOU issue we must do this check after starting the replication stream. We
+    // cannot use the replication client to do that because it's already in CopyBoth mode.
+    // [1] https://www.postgresql.org/docs/15/protocol-replication.html#PROTOCOL-REPLICATION-START-REPLICATION-SLOT-LOGICAL
+    let min_resume_lsn = super::fetch_slot_resume_lsn(&metadata_client, slot).await?;
```

Comment:
Why did this move here?

Reply:
Because I need the `metadata_client` moved into the task, not the async stream